infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -235,17 +235,17 @@ public void prepare(
 
   @Override
   public void cleanup(Workspace workspace) throws InfrastructureException {
-    if (EphemeralWorkspaceUtility.isEphemeral(workspace)) {
-      return;
-    }
-
     AccountImpl account = ((WorkspaceImpl) workspace).getAccount();
     if (isPersonalAccount(account) && accountHasNoWorkspaces(account)) {
       log.debug("Deleting the common PVC: '{}',", configuredPVCName);
       deleteCommonPVC(workspace);
       return;
     }
 
+    if (EphemeralWorkspaceUtility.isEphemeral(workspace)) {
+      return;
+    }
+
```

Comment:
@dkwon17 could we have a test case for this part e.g. if the workspace is ephemeral and account Has No Workspaces the deleteWorkspacePVC will be called?

Reply:
Sounds good, my new commit adds this test: `shouldDeleteCommonPVCIfPersistAttributeIsSetToFalseInWorkspaceConfigAndWorkspacesEmptyWhenCleanupCalled` (maybe the name is too long?)